zerver/tests/test_users.py

```diff
@@ -67,6 +68,21 @@ def test_do_change_is_admin(self) -> None:
         with self.assertRaises(AssertionError):
             do_change_is_admin(user_profile, True, permission='totally-not-valid-perm')
 
+    def test_change_is_primary_admin(self) -> None:
+        iago = self.example_user('iago')
+        hamlet = self.example_user('hamlet')
+
+        self.assertEqual(hamlet.is_primary_admin, False)
+        do_change_is_primary_admin(hamlet, True)
```

Comment:
A few notes on this test:
(1) We should change `populate_db` to make sure Iago is set as the primary administrator in the prebuilt test database.
(2) It'd generally better for testing to hit the API, rather than calling `do_change_...` functions, since then we're testing more of the logic.  So the call on this line should do that.